Io

oddcoder opened the pull request.

oddcoder committed:
adding unit test file

oddcoder committed:
Alligning lines for readability

oddcoder committed:
implement function to get of useless parts

including:
- comments
- empty lines
- spaces at both right and left

shakram02 commented on src/lib/filehandler.rs:
```diff
@@ -1,16 +1,52 @@
+            buf: f,
         };
     }
+    pub fn read_instruction(&mut self) -> Option<Result<String>> {
```
Why don't you just return an `Option<Result,&str>` instead of `Option<Result>` ?

the `&str` might hold the error message, if you want to skip it, just use an option

oddcoder replied:
pls recheck Result type definition 
which is Result<T> it returns T if everything went right and Error if error happened.
I guess &str wont work because its lifetime will end once function returns ?

shakram02 replied:
nope, not necessarily check [this](https://github.com/oddcoder/sickassembler/blob/master/src/lib/pass_two/translator.rs#L37) out, ( in case the error can be statically written in code )
and [this](https://github.com/oddcoder/sickassembler/blob/master/src/lib/basic_types/instruction.rs#L162) (in case the error string isn't determined at compile time)

shakram02 commented on src/lib/filehandler.rs:
```diff
@@ -1,16 +1,52 @@
+        }
+        return Some(Ok(line));
+    }
+    fn scrap_comment (&mut self) -> Option<Result<String>> {
```
same here as above

shakram02 committed:
Reduce complexity in filehandler

shakram02 committed:
Replace `Result<String>` with `Option<String>`

An io error maps to an os error while using the `read_line` function
this error isn't recoverable, so a panic is done in this case.

To denote an EOF, a `None` value is returned, which is more
convenient and clearer

shakram02 committed:
Display OS error to caller

shakram02 committed:
Remove unnecessary comments
